cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -351,35 +364,47 @@ private String getMasterPath() {
   }
 
   private V1PodSpec getPodSpecWithCConf(PodInfo podInfo) {
-    List<V1Volume> volumes = podInfo.getVolumes();
-    List<V1VolumeMount> containerVolumeMounts = podInfo.getContainerVolumeMounts();
-    V1Volume cConfVolume = null;
-    V1VolumeMount cConfVolumeMount = null;
-
-    for (V1VolumeMount containerVolumeMount : containerVolumeMounts) {
+    // podinfo volume
+    List<V1Volume> volumes = new ArrayList<>();
+    V1DownwardAPIVolumeFile labelsFile = new V1DownwardAPIVolumeFile()
+      .fieldRef(new V1ObjectFieldSelector().fieldPath("metadata.labels"))
+      .path(podLabelsFile.getName());
+    V1DownwardAPIVolumeFile nameFile = new V1DownwardAPIVolumeFile()
+      .fieldRef(new V1ObjectFieldSelector().fieldPath("metadata.name"))
+      .path(podNameFile.getName());
+    V1DownwardAPIVolumeFile uidFile = new V1DownwardAPIVolumeFile()
+      .fieldRef(new V1ObjectFieldSelector().fieldPath("metadata.uid"))
+      .path(podUidFile.getName());
+    V1DownwardAPIVolumeSource podinfoVolume = new V1DownwardAPIVolumeSource()
+      .defaultMode(420)
+      .items(Arrays.asList(labelsFile, nameFile, uidFile));
+    String podInfoVolumeName = "podinfo";
+    volumes.add(new V1Volume().downwardAPI(podinfoVolume).name(podInfoVolumeName));
+
+    List<V1VolumeMount> volumeMounts = new ArrayList<>();
+    volumeMounts.add(new V1VolumeMount().name(podInfoVolumeName).mountPath(podInfoDir.getAbsolutePath()));
```

Comment:
should we modify the method name since are also adding volume mounts in this method? Also add a small javadoc?

Reply:
👍  will rename and add javadoc